src/hotspot/cpu/aarch64/c2_CodeStubs_aarch64.cpp

```diff
@@ -95,9 +81,9 @@ void C2HandleAnonOMOwnerStub::emit(C2_MacroAssembler& masm) {
   __ str(rthread, Address(mon, ObjectMonitor::owner_offset_in_bytes()));
 
   // Pop owner object from lock-stack.
-  __ ldr(t, Address(rthread, JavaThread::lock_stack_current_offset()));
+  __ ldr(t, Address(rthread, JavaThread::lock_stack_offset_offset()));
   __ sub(t, t, oopSize);
-  __ str(t, Address(rthread, JavaThread::lock_stack_current_offset()));
+  __ str(t, Address(rthread, JavaThread::lock_stack_offset_offset()));
 
```

Comment:
It looks to me that we should update the '_offset' of LockStack with ldrw, subw and strw instructions here.